File: Python/drmmessages.py
'''
 *
 *	Problem: "DRM Messages"
 *
'''

STARTASCII = 65
NUMLETTERS = 26

'''-----getRotValue-----'''
def getRotValue(input):
	rotValue = 0
	for c in input:
		rotValue += ord(c) - STARTASCII
	rotValue = rotValue % NUMLETTERS
	return rotValue
	
'''-----rotate-----'''
def rotate(input):
	rotValue = getRotValue(input)
	rotated = ""
	for c in input:
		value = ord(c) - STARTASCII
		newValue = STARTASCII + ((value + rotValue) % NUMLETTERS)
		rotated += chr(newValue)
	return(rotated)
	
'''-----merge-----'''
def merge(inputA, inputB):
	merged = ""
	for i in range(len(inputA)):
		c = ord(inputA[i:i+1]) - STARTASCII
		rotateBy = ord(inputB[i:i+1]) - STARTASCII
		newC = ( (c + rotateBy) % NUMLETTERS ) + STARTASCII
		merged += chr(newC)
	return merged
	
'''-----main-----'''
input = input()			

# step 1: divide
lenHalf = len(input) // 2
firstHalf = input[:lenHalf]
secondHalf = input[lenHalf:]

# step 2: rotate
rotatedFirst = rotate(firstHalf)
rotatedSecond = rotate(secondHalf)

# step 3: merge
result = merge(rotatedFirst, rotatedSecond)

print(result)

File: Java/drmmessages.java
/*
 *	Problem: "DRM Messages"
 */

import java.util.Scanner;

public class drmmessages {
	public static int STARTASCII = 65;
	public static int NUMLETTERS = 26;
	
	/*-----getRotValue-----*/
	public static int getRotValue(String input) {
		int rotValue = 0;
		for (int i = 0; i < input.length(); i++) {
			rotValue += (int)input.charAt(i) - STARTASCII;
		}
		rotValue = rotValue % NUMLETTERS;
		return rotValue;
	}
	/*-----rotate-----*/
	public static String rotate(String input) {
		int rotValue = getRotValue(input);
		String rotated = "";
		
		for (int i = 0; i < input.length(); i++) {
			int value = (int)input.charAt(i) - STARTASCII;
			int newValue = STARTASCII + ((value + rotValue) % NUMLETTERS);
			rotated += Character.toString((char) newValue);
		}
		
		return rotated;
	}
	
	/*-----merge-----*/
	public static String merge(String inputA, String inputB) {
		String merged = "";
		for (int i = 0; i < inputA.length(); i++) {
			int c = (int)inputA.charAt(i) - STARTASCII;
			int rotateBy = (int)inputB.charAt(i) - STARTASCII;
			int newC = ( (c + rotateBy) % NUMLETTERS ) + STARTASCII;
			merged += Character.toString(newC);
		}
		return merged;
	}
	
	/*-----main-----*/
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String input = sc.next();
		
		int lenHalf = input.length() / 2;
		String firstHalf = input.substring(0,lenHalf);
		String secondHalf = input.substring(lenHalf);
		
		String rotatedFirst = rotate(firstHalf);
		String rotatedSecond = rotate(secondHalf);
		
		String result = merge(rotatedFirst, rotatedSecond);
		System.out.println(result);
	}	
}